### PR TITLE
[Magic] Clean and add resistance rank logic to resist tiers.

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Adds a limit of 1 to resist tiers when target is very weak to the action element (makink a 1/2 the max resist that will happen)
- Applies level corrected zones to lvl correction logic and makes only mobs gain a boost from lvl correction.
- Applies action bonus macc to base macc.
- Cleans and reorganices `calculateResistRate` function, to make it streamlined and clear.

Pretty much what draft PR #3936 is doing, but cleaner and finished.
Still a great job from Frank.

## Steps to test these changes

No real changes will be notizable at first glanze. Or shouldnt. Cast enfeebling or nuke spells with no errors in log.

### CODE REVIEW

File: scripts/globals/combat/magic_hit_rate.lua [hunk 1]
```diff
@@ -256,30 +263,40 @@ xi.combat.magicHitRate.calculateMagicHitRate = function(magicAcc, magicEva)
...
     if spellElement ~= xi.magic.ele.NONE then
-        resistRank = target:getMod(elementTable[spellElement][4])
+        targetResistRank = target:getMod(elementTable[spellElement][4])
     end
 
-    -- Resistance Ranks "boons".
-    if resistRank > 10 then -- Resistance rank 11 is technically the max, but we check for higher JUST IN CASE something altered it.
-        -- TODO: Inmunobreak logic probably goes here
+    -- Skillchains lowers target resistance rank by 1.
+    local _, skillchainCount = FormMagicBurst(spellElement, target)
```
Comment (WinterSolstice8) [inline scripts/globals/combat/magic_hit_rate.lua:284]: can you add a TODO around here regarding that Rayke needs to be implemented around here? (it lowers resistance rank)

File: scripts/globals/combat/magic_hit_rate.lua [hunk 2]
```diff
@@ -294,17 +311,33 @@ xi.combat.magicHitRate.calculateResistRate = function(actor, target, skillType,
...
-    if resistRank >= 4 then
-        if
-            skillType ~= xi.skill.ELEMENTAL_MAGIC or
-            not actor:hasStatusEffect(xi.effect.SUBTLE_SORCERY)
-        then
-            resistRate = resistRate / 2
-        end
+    -- Force 1/8 if target has max resistance rank.
+    if targetResistRank >= 11 then
+        -- TODO: Inmunobreak logic probably goes here
```
Comment (WinterSolstice8) [inline scripts/globals/combat/magic_hit_rate.lua]: nitpick, not required, `Immunobreak`

**Review by WinterSolstice8 — APPROVED** [review]

File: scripts/globals/combat/magic_hit_rate.lua (hunk 1 above)
Comment (Xaver-DaRed) [reply scripts/globals/combat/magic_hit_rate.lua:284]: I can, however, wouldn't rayke apply a change to the base modifier itself, meaning the mod fetch would already include it?

Comment (WinterSolstice8) [reply scripts/globals/combat/magic_hit_rate.lua:284]: If that seems more appropriate, sure. I don't really mind where it's done, just a TODO for a reminder

Comment (Xaver-DaRed) [reply scripts/globals/combat/magic_hit_rate.lua:284]: I added the TODO anyway.

It all depends on how Ryake is implemented anyway, but the TODO can't hurt

File: scripts/globals/combat/magic_hit_rate.lua (hunk 2 above)
Comment (Xaver-DaRed) [reply scripts/globals/combat/magic_hit_rate.lua]: oops

File: scripts/globals/combat/magic_hit_rate.lua [hunk 3]
```diff
@@ -256,30 +263,42 @@ xi.combat.magicHitRate.calculateMagicHitRate = function(magicAcc, magicEva)
...
+    local _, skillchainCount = FormMagicBurst(spellElement, target)
+
+    if skillchainCount > 0 then
+        targetResistRank = targetResistRank - 1
     end
 
-    -- Resistance Ranks "boons".
-    if resistRank > 10 then -- Resistance rank 11 is technically the max, but we check for higher JUST IN CASE something altered it.
-        -- TODO: Inmunobreak logic probably goes here
+    -- TODO: Ryake logic might be needed here, depending on how it's implemented.
```
Comment (WinterSolstice8) [inline scripts/globals/combat/magic_hit_rate.lua]: hehe, needed more coffee? `Rayke`

Comment (Xaver-DaRed) [reply scripts/globals/combat/magic_hit_rate.lua]: It does seem like I do

**Pull request merged**